### PR TITLE
add diary edit page, diary calories test

### PR DESCRIPTION
**Pull request by astiuk**

### CODE REVIEW

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/mfp/pages/android/DiaryPage.java [hunk 1]
```diff
@@ -13,12 +17,56 @@ public class DiaryPage extends DiaryPageBase {
...
+    public boolean isDiaryClean() {
+        return !editDiaryButton.isPresent(1);
+    }
+
+    @Override
+    public void clearDiary() {
+        if(!isPageOpened())
+        {
+            CommonPageBase commonPage = initPage(getDriver(), CommonPageBase.class);
+            commonPage.clickBottomNavigatorButton(BottomNavigatorButtons.DIARY);
```
Comment (I-Iemelianova) [inline src/main/java/com/qaprosoft/carina/demo/mobile/gui/mfp/pages/android/DiaryPage.java]: wrong logic - we shoul call this method ONLY from Diary page, all this structure should be deleted

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/mfp/pages/android/DiaryPage.java [hunk 2]
```diff
@@ -13,12 +17,56 @@ public class DiaryPage extends DiaryPageBase {
...
+        }
+        if(!isDiaryClean())
+        {
+            editDiaryButton.click();
```
Comment (I-Iemelianova) [inline src/main/java/com/qaprosoft/carina/demo/mobile/gui/mfp/pages/android/DiaryPage.java]: pls, add timeout for click method

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/mfp/pages/android/EditDiaryPage.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    @FindBy(id = "com.myfitnesspal.android:id/select_all")
+    private ExtendedWebElement selectAllCheckbox;
+
+    public EditDiaryPage(WebDriver driver) {
+        super(driver);
+    }
+
+    @Override
+    public ConfirmDeletePageBase clickTrashBin() {
+        trashBinButton.click();
```
Comment (I-Iemelianova) [inline src/main/java/com/qaprosoft/carina/demo/mobile/gui/mfp/pages/android/EditDiaryPage.java]: timout for click

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/mfp/pages/android/ConfirmDeletePage.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+import com.qaprosoft.carina.core.foundation.utils.factory.DeviceType;
+import com.qaprosoft.carina.core.foundation.webdriver.decorator.ExtendedWebElement;
+import com.qaprosoft.carina.demo.mobile.gui.mfp.pages.common.ConfirmDeletePageBase;
+import com.qaprosoft.carina.demo.mobile.gui.mfp.pages.common.DiaryPageBase;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.FindBy;
+
+@DeviceType(pageType = DeviceType.Type.ANDROID_PHONE, parentClass = ConfirmDeletePageBase.class)
+public class ConfirmDeletePage extends ConfirmDeletePageBase {
```
Comment (I-Iemelianova) [inline src/main/java/com/qaprosoft/carina/demo/mobile/gui/mfp/pages/android/ConfirmDeletePage.java]: this isn't a standalone page, ConfirmDelete modal is a part of EditDiaryPage, no need to create more page objects that needed

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/mfp/pages/android/ConfirmDeletePage.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    @FindBy(xpath = "//android.widget.Button[@text='Delete']")
+    private ExtendedWebElement deleteButton;
+
+    public ConfirmDeletePage(WebDriver driver) {
+        super(driver);
+    }
+
+    @Override
+    public DiaryPageBase clickDelete() {
+        deleteButton.click();
```
Comment (I-Iemelianova) [inline src/main/java/com/qaprosoft/carina/demo/mobile/gui/mfp/pages/android/ConfirmDeletePage.java]: timout

File: src/test/java/com/qaprosoft/carina/demo/mfp/MobileTest.java [hunk 6]
```diff
@@ -40,4 +40,18 @@ public void bottomNavigationButtonsTest() {
...
+
+    @Test
+    @MethodOwner(owner = "Hostiuk")
+    @TestRailCases(testCasesId = "2")
+    @TestLabel(name = "feature", value = {"mobile", "regression"})
+    public void goalCaloriesEqualsRemainCaloriesTest() {
+        LoginPageBase loginPage = initPage(getDriver(), LoginPageBase.class);
+        loginPage.login(R.TESTDATA.get("email"), R.TESTDATA.get("password"));
+
+        DiaryPageBase diaryPage = initPage(getDriver(), DiaryPageBase.class);
```
Comment (I-Iemelianova) [inline src/test/java/com/qaprosoft/carina/demo/mfp/MobileTest.java]: here you should add opening Diary page, as it was said earlier - you can't put it in the method in this case.

File: src/test/java/com/qaprosoft/carina/demo/mfp/MobileTest.java [hunk 7]
```diff
@@ -40,4 +40,18 @@ public void bottomNavigationButtonsTest() {
...
+        diaryPage.clearDiary();
+        Assert.assertEquals(diaryPage.getGoalCalories(), diaryPage.getRemainCalories(),
+                "Goal calories and remain calories aren't equal");
+    }
```
Comment (I-Iemelianova) [inline src/test/java/com/qaprosoft/carina/demo/mfp/MobileTest.java:57]: Good job!

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/mfp/pages/android/ConfirmDeletePage.java (hunk 4 above)
Comment (astiuk) [reply src/main/java/com/qaprosoft/carina/demo/mobile/gui/mfp/pages/android/ConfirmDeletePage.java]: So, is it better to move ConfirmDeletePage class inside EditDiaryPage class, or just move methods and locators?

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/mfp/pages/android/EditDiaryPage.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+        trashBinButton.click(3);
+    }
+
+    @Override
+    public void checkSelectAll() {
+        selectAllCheckbox.check();
+    }
+
+    @Override
+    public DiaryPageBase clickConfirmDelete() {
```
Comment (I-Iemelianova) [inline src/main/java/com/qaprosoft/carina/demo/mobile/gui/mfp/pages/android/EditDiaryPage.java]: too many methods, it's not right to create 3 methods for one needed action

**Pull request merged**